nemo_run/core/execution/skypilot.py

```diff
@@ -361,20 +376,48 @@ def to_task(
 head_node_ip=`echo "$SKYPILOT_NODE_IPS" | head -n1`
 echo "head_node_ip=$head_node_ip"
 
+export MASTER_ADDR=$head_node_ip
+export WORLD_SIZE=$num_nodes
+export RANK=`echo "$SKYPILOT_NODE_RANK"`
+echo "MASTER_ADDR=$MASTER_ADDR"
```

Comment:
I don't think you need these as they are passed explicitly in the generated `torchrun` command.